### PR TITLE
Adds a Layer component for showing geojson

### PR DESCRIPTION
**Pull request by paulpostel**
[Card](https://rentpath.leankit.com/card/646884884)

### CODE REVIEW

File: packages/react-ui-core/src/Gmap/Layer.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  componentWillUnmount() {
+    this.removeFeatures()
+  }
+
+  removeFeatures() {
+    const { map } = this.props
+
+    this.features.forEach(feature => {
+      if (map) map.data.remove(feature)
+      this.features.delete(feature)
```
Comment (tcwilson) [inline packages/react-ui-core/src/Gmap/Layer.js]: It's probably slightly more efficient to do `this.features.clear()` after the `forEach`. But if this set only has a few hundred items it probably doesn't matter much.

**Comment by tcwilson** [comment]
👍

File: packages/react-ui-core/src/Gmap/Layer.js (hunk 1 above)
Comment (tadjohnston) [reply packages/react-ui-core/src/Gmap/Layer.js]: If it's ever possible it would have more than that, I'd prefer the clear

**Comment by tadjohnston** [comment]
@paulpostel here is mine that was gonna create for ag.js map markers: https://gist.github.com/tadjohnston/1c250234245e343a7b333302ffc19076

If you see anything in there you might want to use, feel free to add it. The hope was that the GeojsonMarkers I was creating would eventually make it into react-ui-core.

**Comment by paulpostel** [comment]
@tcwilson pointed out that the `data.setStyle` function would accept a function argument that receives the feature-to-be-styled.  That means I wouldn't have to `data.overrideStyle` for each feature in the `Layer`.  since the styling function is one-per-map, what do you think about adding it as an optional prop to `Gmap`?  I guess it could also be set thru a separate `GmapData` component or something.

File: packages/react-ui-core/src/Gmap/Layer.js (hunk 1 above)
Comment (paulpostel) [reply packages/react-ui-core/src/Gmap/Layer.js]: agreed, `clear()` is better, i'll update

**Comment by paulpostel** [comment]
@tadjohnston, @tcwilson I added the styling function to `Gmap` as described above.  I think it's a better way to go.  Please take a another look at this.

**Pull request merged**